mitodl/open-edx-plugins
# Added Tutor in CI

### What are the relevant tickets?
https://github.com/mitodl/hq/issues/4443

### Description (What does it do?)
This PR replaces devstack with tutor in the CI for initializing edx-platform and running the integration tests of plugins. More details in the [ticket](https://github.com/mitodl/hq/issues/4443)

### How can this be tested?
- Please see the CI job logs
- To test locally, try to follow the job steps from the CI on your local machine.( this will take a while to run )
   - Build all the plugins packages using `pants build ::`
   - install tutor:
       - Nightly for Master Branch
          ```
           git clone --branch=nightly https://github.com/overhangio/tutor.git
           pip install -e "./tutor"
           ```
       - For Quince branch `pip install tutor==17.0.6`
       - For Redwood branch `pip install tutor==18.0.0`
   - Go to edx-platform directory and run `tutor mounts add .`
   - Build tutor images by running `tutor images build openedx-dev`
   - Run the launch command of tutor to complete the setup `tutor dev launch --non-interactive`
   - Stop the running containers `tutor dev stop`
   - Run the following commands:
       ```
        DEV="tutor_dev"  #"tutor_nightly_dev" in case of master branch
        docker-compose -f $(tutor config printroot)/env/local/docker-compose.yml -f $(tutor config printroot)/env/dev/docker-compose.yml --project-name $DEV run -v $PWD/../open-edx-plugins:/open-edx-plugins lms /open-edx-plugins/run_devstack_integration_tests.sh       
       ```
   - Validate the tests running and passing successfully
